src/components/code/code_block.tsx

```diff
@@ -388,9 +398,9 @@ const useFullScreen = ({
 }) => {
   const [isFullScreen, setIsFullScreen] = useState(false);
 
-  const toggleFullScreen = () => {
+  const toggleFullScreen = useCallback(() => {
     setIsFullScreen(!isFullScreen);
-  };
+  }, [isFullScreen]);
```

Comment:
optional perf improvement
```suggestion
  const toggleFullScreen = useCallback(() => {
    setIsFullScreen((isFullScreen) => !isFullScreen);
  }, []);
```